### PR TITLE
"SHOULD" fixes based on AD review and other input

### PR DESCRIPTION
**Pull request by bradleypeabody**
Here are changes to address the "SHOULD" concerns based on the review done discussion in #114

### CODE REVIEW

File: draft-ietf-uuidrev-rfc4122bis.md [hunk 1]
```diff
@@ -1296,9 +1296,10 @@ Truncating:
 
 Error Handling:
 : If a system overruns the generator by requesting too many UUIDs
-  within a single system time interval, the UUID service SHOULD either
+  within a single system time interval, the UUID service can either
   return an error, or stall the UUID generator until the system clock
-  catches up, and MUST NOT return knowingly duplicate values.
+  catches up, and MUST NOT return knowingly duplicate values due to
+  counter rollover.
```
Comment (bradleypeabody) [inline draft-ietf-uuidrev-rfc4122bis.md:1302]: @LiosK I saw your comment on #114, I think this works to keep it MUST NOT but specifically say "due to counter rollover", so it excludes other edge cases such as the clock rolling back. I think that aligns with the intention of the text anyway, which is all about counters.

File: draft-ietf-uuidrev-rfc4122bis.md [hunk 2]
```diff
@@ -1664,8 +1666,9 @@ Implementations MAY implement a shared knowledge scheme introduced in {{distribu
 ## Unguessability {#unguessability}
 Implementations SHOULD utilize a cryptographically secure pseudo-random number
 generator (CSPRNG) to provide values that are both difficult to predict ("unguessable")
-and have a low likelihood of collision ("unique").
-Care SHOULD be taken to ensure the CSPRNG state is properly reseeded upon
+and have a low likelihood of collision ("unique").  The exception is when a
+suitable CSPRNG is unavailable in the execution environment.
+Take care to ensure the CSPRNG state is properly reseeded upon
```
Comment (bradleypeabody) [inline draft-ietf-uuidrev-rfc4122bis.md:1671]: I think stating the intended exception here (no CSPRNG available) makes sense and clearly justifies the use of SHOULD.

File: draft-ietf-uuidrev-rfc4122bis.md [hunk 3]
```diff
@@ -1296,9 +1296,10 @@ Truncating:
 
 Error Handling:
 : If a system overruns the generator by requesting too many UUIDs
-  within a single system time interval, the UUID service SHOULD either
+  within a single system time interval, the UUID service can either
```
Comment (cbandy) [inline draft-ietf-uuidrev-rfc4122bis.md:1299]: Are only two possible actions allowed here, error or stalling? If not, I recommend removing the word "either" here.

Paraphrasing: If too many, maybe error or maybe stall; DO NOT duplicate.

File: draft-ietf-uuidrev-rfc4122bis.md [hunk 4]
```diff
@@ -1428,9 +1430,9 @@ Fixed-Length Dedicated Counter Length:
   the level of timestamp precision in use.
   For example, millisecond precision generally requires a larger counter than a
   timestamp with nanosecond precision.
-  General guidance is that the counter SHOULD be at least 12 bits but no longer
+  General guidance is that the counter should be at least 12 bits but no longer
```
Comment (LiosK) [inline draft-ietf-uuidrev-rfc4122bis.md:1433]: How about: "General guidance is that the counter SHOULD be at least 12 bits but no longer than 42 bits for UUIDv7."? This guidance isn't really effective without the v7 context, and with that context BCP 14 verbiage makes sense.

**Pull request merged**